File: java/DataCounter.java
import java.util.List;
import java.util.Map;
import java.util.HashMap;

public class DataCounter {
    public static Map<String, Object> jsonFormatCounters(List<Map<String, Object>> data, int page) {
        Map<String, Object> result = new HashMap<>();

        if (data == null || data.isEmpty()) {
            result.put("data", data);
            result.put("totalCount", 0);
            result.put("itemCount", 0);
            result.put("page", page);
            return result;
        }

        int itemCount = data.size();
        int totalCount = data.stream()
                             .mapToInt(object -> Integer.parseInt(object.get("count").toString()))
                             .sum();

        result.put("data", data);
        result.put("totalCount", totalCount);
        result.put("itemCount", itemCount);
        result.put("page", page);

        return result;
    }
}

File: clojure/data-counter.clj
(defn parse-int [s]
  (try
    (Integer/parseInt s)
    (catch NumberFormatException e 0)))

(defn json-format-counters [data page]
  (if (or (nil? data) (empty? data))
    {:data [] :totalCount 0 :itemCount 0 :page page}
    (let [item-count (count data)
          total-count (reduce + (map #(parse-int (% "count")) data))]
      {:data data :totalCount total-count :itemCount item-count :page page})))